### PR TITLE
Add ranges utitlities to math

### PR DESCRIPTION
**Pull request by cypherdare**
This is for #227.

### CODE REVIEW

File: math/src/main/kotlin/ktx/math/ranges.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ * range.
+ *
+ * This function uses `MathUtils.randomTriangular(min, max, mode)`. Note that [endInclusive][ClosedRange.endInclusive]
+ * is treated as exclusive as it is not practical to keep it inclusive. Results are undefined for an empty range, and
+ * there is no error checking.
+ */
+fun ClosedRange<Float>.randomTriangular(normalizedMode: Float) =
+    MathUtils.randomTriangular(start, endInclusive,
+        normalizedMode * (endInclusive - start) + start
+    )
```
Comment (czyzby) [inline math/src/main/kotlin/ktx/math/ranges.kt]: Missing new line.

File: math/src/test/kotlin/ktx/math/rangesTest.kt [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  fun `should subtract from int range`() {
+    val original = -20..500
+    val subtrahend = 37
+    val shifted = original - subtrahend
+    assertEquals(original.first - subtrahend, shifted.first)
+    assertEquals(original.last - subtrahend, shifted.last)
+  }
+
+  @Test
+  fun `should produce expected random int values`() {
```
Comment (czyzby) [inline math/src/test/kotlin/ktx/math/rangesTest.kt]: Note sure about this test. It seems to go into too much detail about how the internal implementation works. For tests such as this, I think I'd repeat the `range.random` call, save the results and check:
- If every value is within the range.
- If the values (roughly) produce uniform distribution.

File: math/src/test/kotlin/ktx/math/rangesTest.kt [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+  fun `should subtract from float range`() {
+    val original = -20f..500f
+    val subtrahend = 37f
+    val shifted = original - subtrahend
+    assertEquals(original.start - subtrahend, shifted.start)
+    assertEquals(original.endInclusive - subtrahend, shifted.endInclusive)
+  }
+
+  @Test
+  fun `should produce expected random float values`() {
```
Comment (czyzby) [inline math/src/test/kotlin/ktx/math/rangesTest.kt]: Same goes for this test. You can include an external math library in test scopes if that makes the asserts any easier.

File: math/src/test/kotlin/ktx/math/rangesTest.kt [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+              val probability = 1f - leftTriangleArea - rightTriangleArea
+              checkRange to probability
+            }
+    for ((checkRange, probability) in rangesToExpectedProbabilities) {
+      var result = values.count { it in checkRange }.toFloat() / count
+      assert(abs(result - probability) / probability <= allowableError)
+    }
+  }
+
+}
```
Comment (czyzby) [inline math/src/test/kotlin/ktx/math/rangesTest.kt]: Missing new line.

**Comment by cypherdare** [comment]
Good points, updated!

**Comment by czyzby** [comment]
Thanks for the contribution.

**Pull request merged**